src/conversion.c

```diff
@@ -341,26 +353,55 @@ int igraph_get_edgelist(const igraph_t *graph, igraph_vector_t *res, igraph_bool
 
 int igraph_to_directed(igraph_t *graph,
                        igraph_to_directed_t mode) {
-
-    if (mode != IGRAPH_TO_DIRECTED_ARBITRARY &&
-        mode != IGRAPH_TO_DIRECTED_MUTUAL) {
-        IGRAPH_ERROR("Cannot direct graph, invalid mode", IGRAPH_EINVAL);
-    }
+    long int no_of_edges = igraph_ecount(graph);
+    long int no_of_nodes = igraph_vcount(graph);
 
     if (igraph_is_directed(graph)) {
-        return 0;
-    }
-
-    if (mode == IGRAPH_TO_DIRECTED_ARBITRARY) {
-
+        return IGRAPH_SUCCESS;
+    }   
+
+    switch (mode) {
+    case IGRAPH_TO_DIRECTED_ARBITRARY:
+    case IGRAPH_TO_DIRECTED_RANDOM:
+    case IGRAPH_TO_DIRECTED_ACYCLIC:
+      {
         igraph_t newgraph;
         igraph_vector_t edges;
-        long int no_of_edges = igraph_ecount(graph);
-        long int no_of_nodes = igraph_vcount(graph);
         long int size = no_of_edges * 2;
+        long int i;
+
         IGRAPH_VECTOR_INIT_FINALLY(&edges, size);
         IGRAPH_CHECK(igraph_get_edgelist(graph, &edges, 0));
 
+        if (mode == IGRAPH_TO_DIRECTED_RANDOM) {
```

Comment:
It is a bit strange to have this `if` inside a `switch` statement on the exact same variable. Why not put this under the `IGRAPH_TO_DIRECTED_RANDOM` `case`?

Reply:
Almost all code is shared between the ARBITRARY / ACYCLIC / RANDOM cases. The only difference is that we do some special ordering for the ACYCLIC / RANDOM cases in the middle of the implementation.

If I used 4 different `case`s in switch (as I originally planned to—this is why I changed the original `if` to a `switch`) then 3 of those cases would be almost identical, duplicating a lot of code.

Reply:
Yes, that is right. But couldn't we then also reduce code duplication between the last `case`? In principle it is the same, except that the edges are included twice. It is now a bit strange mix of `if` and `switch`, that when reading back might seem a bit counterintuitive. If you want I can also do this.

Reply:
That case is different enough that I think that entangling it with the first case would hurt readability too much.